monkey/monkey_island/cc/environment/utils.py

```diff
@@ -44,8 +43,60 @@ def _create_secure_directory_windows(path: str):
             windows_permissions.get_security_descriptor_for_owner_only_perms()
         )
         win32file.CreateDirectory(path, security_attributes)
+
+    except Exception as ex:
+        LOG.error(f'Could not create a directory at "{path}": {str(ex)}')
+        raise ex
+
+
+def create_secure_file(path: str):
+    if not os.path.isfile(path):
+        if is_windows_os():
+            _create_secure_file_windows(path)
+        else:
+            _create_secure_file_linux(path)
+
+
+def _create_secure_file_linux(path: str):
+    try:
+        flags = (
+            os.O_RDWR | os.O_CREAT | os.O_EXCL
+        )  # read/write, create new, throw error if file exists
+        mode = 0o700  # read/write/execute permissions to owner
+        os.close(os.open(path, flags, mode))
+
     except Exception as ex:
-        LOG.error(
-            f'Could not create a directory at "{path}": {str(ex)}'
+        LOG.error(f'Could not create a file at "{path}": {str(ex)}')
+        raise ex
+
+
+def _create_secure_file_windows(path: str):
+    try:
+        file_access = win32file.GENERIC_READ | win32file.GENERIC_WRITE
+        file_sharing = (
+            win32file.FILE_SHARE_READ
+        )  # subsequent open operations on the object will succeed only if read access is requested
+        security_attributes = win32security.SECURITY_ATTRIBUTES()
+        security_attributes.SECURITY_DESCRIPTOR = (
+            windows_permissions.get_security_descriptor_for_owner_only_perms()
+        )
+        file_creation = win32file.CREATE_NEW  # fails if file exists
+        file_attributes = win32file.FILE_FLAG_BACKUP_SEMANTICS
+
+        win32file.CloseHandle(
+            win32file.CreateFile(
+                path,
+                file_access,
+                file_sharing,
+                security_attributes,
+                file_creation,
+                file_attributes,
+                win32job.CreateJobObject(
+                    None, ""
+                ),  # https://stackoverflow.com/questions/46800142/in-python-with-pywin32-win32job-the-createjobobject-function-how-do-i-pass-nu  # noqa: E501
```

Comment:
No clue about what `win32job.CreateJobObject` does, why do we call it here, what argument it specifies etc. The whole `win32file.CreateFile` function should be called with keyword arguments.

Reply:
Generally, just looks like a dirty workaround, there should be an easier  way to pass `Null`. Either way, this should be hidden in a clearly named function.

Reply:
> The whole `win32file.CreateFile` function should be called with keyword arguments.

Can't. It throws an error that it can't be called with keyword arguments.

> Generally, just looks like a dirty workaround, there should be an easier way to pass `Null`.

I agree. Couldn't find an easier way, though. I'll try again, perhaps.

Reply:
It's OK if that's the only way we can provide `NULL` argument, but you should create a method called `get_null_value_for_win32` or something and call it instead. Then it would be clear that we actually provide `NULL` as the last argument and not some kind of created job.

Lastly, don't put comments that obviously violate line length inline, put them above the statement.